lib/ansible/modules/cloud/amazon/cloudfront_facts.py

```diff
@@ -0,0 +1,448 @@
+    invalidation = module.params.get('invalidation')
+    streaming_distribution = module.params.get('streaming_distribution')
+    streaming_distribution_config = module.params.get('streaming_distribution_config')
+
+    list_cloud_front_origin_access_identities = module.params.get('list_cloud_front_origin_access_identities')
+    list_distributions = module.params.get('list_distributions')
+    list_distributions_by_web_acl_id = module.params.get('list_distributions_by_web_acl_id');
+    list_invalidations = module.params.get('list_invalidations')
+    list_streaming_distributions = module.params.get('list_streaming_distributions')
+
+    require_distribution_id = (distribution or distribution_config or invalidation or
+            streaming_distribution or streaming_distribution_config or list_invalidations)
+
+    # validations
+    if require_distribution_id and distribution_id is None and domain_name_alias is None:
+        module.fail_json(msg='Error distribution_id or domain_name have not been specified.')
+    if (invalidation and invalidation_id is None):
+        module.fail_json(msg='Error invalidation_id has not been specified.')
+    if (cloud_front_origin_access_identity or cloud_front_origin_access_identity_config) and cloud_front_origin_access_identity_id is None:
+        module.fail_json(msg='Error cloud_front_origin_access_identity_id has not been specified.')
+    if list_distributions_by_web_acl_id and web_acl_id is None:
+        module.fail_json(msg='Error web_acl_id has not been specified.')
+
+    # get distribution id from domain name alias
+    if require_distribution_id and distribution_id is None:
+        distribution_id = service_mgr.get_distribution_id_from_domain_name(domain_name_alias)
+        if not distribution_id:
+            module.fail_json(msg='Error unable to source a distribution id from domain_name_alias')
+
+    # set appropriate ansible_facts id
+    if distribution_id:
+        result = { 'ansible_facts': { 'cloudfront': { distribution_id:{} } } }
+        facts = result['ansible_facts']['cloudfront'][distribution_id]
+    elif cloud_front_origin_access_identity_id:
+        result = { 'ansible_facts': { 'cloudfront': { cloud_front_origin_access_identity_id:{} } } }
+        facts = result['ansible_facts']['cloudfront'][cloud_front_origin_access_identity_id]
+    elif web_acl_id:
+        result = { 'ansible_facts': { 'cloudfront': { web_acl_id:{} } } }
+        facts = result['ansible_facts']['cloudfront'][web_acl_id]
+    else:
+        result = { 'ansible_facts': { 'cloudfront': {} } }
+        facts = result['ansible_facts']['cloudfront']
+
+    # call details based on options
+    if distribution:
+        facts['distribution'] = service_mgr.get_distribution(distribution_id)
+    if distribution_config:
+        facts['distribution_config'] = service_mgr.get_distribution_config(distribution_id)
+    if cloud_front_origin_access_identity:
```

Comment:
Perhaps you could drop the `cloud_front_` prefix for methods and variables, since everything refers to Cloudfront properties? Unless of course it is has something to do with being consistent with the corresponding AWS API response.